Develop

vadimnet opened the pull request:
Добавлены основные маршруты и создана структура приложения

vadimnet committed:
Add files via upload

vadimnet committed:
Все еще тест

vadimnet committed:
Начало

vadimnet committed:
Update README.md

vadimnet committed:
Начата разработка структуры приложения

vadimnet committed:
Merge branch 'develop' of https://github.com/vadimnet/FastAPI-project into develop

vadimnet committed:
Разработка структуры приложения

vadimnet committed:
Переделана структура приложения

vadimnet committed:
Delete __pycache__ directory

vadimnet committed:
Добавлены: маршруты и async функции без содержимо-
го, одна pydantic-модель для создания аккаунта,добавлена
папка models, которая будет содержать все pydantic-
-модели.

vadimnet committed:
Merge branch 'develop' of https://github.com/vadimnet/FastAPI-project into develop

vadimnet committed:
Добавлены два маршурта: "logout" и "account"; для
выхода из аккаунта и получения страницы своего аккаунта

vadimnet committed:
Добавлен  маршрут account с методом put для изме-
нения данных аккаунта клиента

shekshuev commented on app.py:
```diff
@@ -0,0 +1,14 @@
+from typing import Union
+from fastapi import FastAPI
+
+app = FastAPI()
```
отключи тут cors

shekshuev commented on auth/models/User.py:
```diff
@@ -0,0 +1,6 @@
+class User(BaseModel):
+    token: str
+    username: str
+    password_hash: str
```
пароль хэшируется на стороне бэкенда. Оптимально принимать поля password и password_confirmation, сравнивать их, если ок,  то хэшируем и в БД уходит уже хэш пароля

shekshuev commented on routers/login.py:
```diff
@@ -0,0 +1,9 @@
+    prefix="/api/v1.0/login"
+)
+
+@router.put("/api/v1.0/login")
```
это post должно быть

shekshuev commented on main.py:
```diff
@@ -1,5 +1,8 @@
-from routers import login
+from routers import login, accounts
 
 app = FastAPI()
```
указывал ранее, отключи cors

shekshuev commented on models/CreateAccount.py:
```diff
@@ -0,0 +1,10 @@
+
+class CreateAccount(BaseModel):
+    login: str
+    password: str
```
см. ранее, добавь еще password_confirmation

shekshuev commented on models/CreateAccount.py:
```diff
@@ -0,0 +1,10 @@
+from pydantic import BaseModel
+
+
+class CreateAccount(BaseModel):
```
такие классы называтются data transfer object - DTO. В твоем случае CreateAccountDTO. Класс, который ты будешь отдавать - ReadAccountDTO. Для обновления UpdateAccountDTO и т.д.

shekshuev commented on routers/accounts.py:
```diff
@@ -0,0 +1,23 @@
+async def accounts():
+    pass
+
+@router.put("accounts")
```
метод delete тут, а id для удаления должен быть в урле accounts/1

shekshuev commented on routers/accounts.py:
```diff
@@ -0,0 +1,23 @@
+async def delete_account(id: Annotated[int, Query(gt=0)]):
+    pass
+
+@router.put("accounts")
```
тут post

shekshuev commented on routers/accounts.py:
```diff
@@ -0,0 +1,23 @@
+    pass
+
+@router.put("accounts/{id}")
+async def change_account(id: int, account:CreateAccount): #сделать отдельную pydantic модель?(или просто сделать текущую универсальной)
```
да, отдельная модель, UpdateAccountDTO

shekshuev commented on routers/login.py:
```diff
@@ -1,9 +1,15 @@
 )
 
-@router.put("/api/v1.0/login")
+@router.put("login")
```
тут post

shekshuev commented on routers/login.py:
```diff
@@ -1,9 +1,15 @@
     pass
+
+@router.put("sign-up")
+async def sign_up():
```
тут post. И обычно используется пара register/login или sign-up/sign-in

shekshuev commented on routers/accounts.py:
```diff
@@ -20,4 +20,8 @@ async def create_account(account: CreateAccount): #попробовать сде
+    pass
+
+@router.get("account")
+async def account(id: Annotated[int, Query()]):
```
название семантически неверное. get_account / get_accounts. Аналогично с delete, create и т.д. плюс тут id не квери параметр, а path

shekshuev commented on routers/accounts.py:
```diff
@@ -25,3 +25,7 @@ async def change_account(id: int, account:CreateAccount): #сделать отд
 async def account(id: Annotated[int, Query()]):
     pass
+
+@router.put("account")
```
тут id аккаунта в path должен лежать

vadimnet committed:
Исправления
Добавлены новые модели, добавлена валидация полей моделей, исправлены методы в путях, отключен CORS

vadimnet committed:
Исправления
Добавлены описания для всех полей

vadimnet committed:
Исправления
